web/src/settings_org.js

```diff
@@ -704,29 +708,191 @@ export function discard_property_element_changes(elem, for_realm_default_setting
     update_dependent_subsettings(property_name);
 }
 
+export function valid_to(deletes_in) {
+    const time_valid = Number.parseFloat(deletes_in);
+    if (time_valid === 0) {
+        return $t({defaultMessage: "Data will be deleted immediately"});
+    }
+
+    if (!time_valid) {
+        return $t({defaultMessage: "Data will not be automatically deleted"});
+    }
+
+    // The below is a duplicate of timerender.get_full_datetime, with a different base string.
+    const valid_to = add(new Date(), {minutes: time_valid});
+    const date = timerender.get_localized_date_or_time_for_format(valid_to, "dayofyear_year");
+    const time = timerender.get_localized_date_or_time_for_format(valid_to, "time");
+
+    return $t({defaultMessage: "Data to be deleted on {date} at {time}"}, {date, time});
+}
+
 export function deactivate_organization(e) {
     e.preventDefault();
     e.stopPropagation();
 
     function do_deactivate_realm() {
+        let deletes_in = $("#deletes_in").val();
+        // See settings_config.realm_deletion_in_values for why we do this conversion.
+        if (deletes_in === "null") {
+            deletes_in = JSON.stringify(null);
+        } else if (deletes_in === "custom") {
+            deletes_in = Number.parseFloat(
+                settings_ui.get_custom_input_time_in_minutes(
+                    custom_deletion_time_unit,
+                    custom_deletion_time_input,
+                ),
+            );
+        } else {
+            deletes_in = Number.parseFloat($("#deletes_in").val());
+        }
+        const data = {
+            realm_deletion_in_minutes: deletes_in,
+        };
         channel.post({
             url: "/json/realm/deactivate",
+            data,
             error(xhr) {
                 ui_report.error($t_html({defaultMessage: "Failed"}), xhr, $("#dialog_error"));
             },
         });
     }
 
-    const html_body = render_settings_deactivate_realm_modal();
+    const time_unit_choices = ["days", "weeks"];
+    const minimum_allowed_days = page_params.min_deactivated_realm_deletion_days;
+    const maximum_allowed_days = page_params.max_deactivated_realm_deletion_days;
+
+    const delete_in_options = {};
+
+    function get_allowed_duration_message() {
+        if (maximum_allowed_days === null) {
+            return $t(
+                {
+                    defaultMessage:
+                        "Enter a value greater than or equal to {minimum_allowed_days} days",
+                },
+                {minimum_allowed_days},
+            );
+        }
+        return $t(
+            {
+                defaultMessage:
+                    "Enter a value between {minimum_allowed_days} and {maximum_allowed_days} days",
```

Comment:
I don't love having two new strings here; it seems like it'd be better to just use the same strings we have in the server, which I've changed to be, for example, `"{field_name} must be at most {max_allowed_days} days."`

Reply:
I think we should include values of both `max_allowed_days` and `min_allowed_days` in this string. This string is not an error message, it will be shown when user is entering a custom time like in screenshot below. 
![Screenshot from 2023-07-06 19-31-14](https://github.com/zulip/zulip/assets/99073049/7ae6e780-6a1a-42ca-97aa-8f59124b1786)

Reply:
The problem with that approach is if one of those two values is `None`, the strings get very awkward -- "Enter a value between None and 100 days" looks bad.